src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -251,6 +249,7 @@ private sealed class RunningInstance
                 _includeSubdirectories = includeSubdirectories;
                 _notifyFilters = notifyFilters;
                 _cancellationToken = cancellationToken;
+                FileSystemWatcher.CaseSensitive = true;
```

Comment:
I made it a property on the base class since, technically, Windows and NTFS supports POSIX compliant-case-sensitivity for apps that support it (https://support.microsoft.com/en-us/kb/100625). 

I believe you can also do some customization on Linux platforms to support case insensitivity, so to support all these potential scenarios, this seemed like the best option. I can change it back to constants if you'd like